io/exportModel.m

```diff
@@ -113,7 +113,13 @@ function exportModel(model,fileName,exportGeneComplexes,supressWarnings)
 model.mets=regexprep(model.mets,'([^0-9_a-zA-Z])','__${num2str($1+0)}__');
 model.comps=regexprep(model.comps,'([^0-9_a-zA-Z])','__${num2str($1+0)}__');
 if isfield(model,'genes')
-    model.genes=regexprep(model.genes,'([^0-9_a-zA-Z])','__${num2str($1+0)}__');
+    problemGenes=find(~cellfun('isempty',regexp(model.genes,'([^0-9_a-zA-Z])')));
+    originalGenes=model.genes(problemGenes);
+    replacedGenes=regexprep(model.genes(problemGenes),'([^0-9_a-zA-Z])','__${num2str($1+0)}__');
+    model.genes(problemGenes)=replacedGenes;
+    for i=1:numel(problemGenes)
+        model.grRules=strrep(model.grRules,originalGenes{i},replacedGenes{i});

```

Comment:
Similar situation here. You are using `strrep` to replace gene names/IDs, which will cause problems with partial matches. I would suggest the following:

`model.grRules = regexprep(model.grRules, ['(^|\s|\()' originalGenes{i} '($|\s|\))'], ['$1' replacedGenes{i} '$2']);`